### PR TITLE
Refuse to create empty commits and other fixes

### PR DESCRIPTION
**Pull request by personalizedrefrigerator**
## Summary
 * Refuses to create empty commits
 * Adds `lg2 rm --cached` (a warning is printed if `--cached` isn't given -- it doesn't touch the working tree)
 * Adds `lg2 add --force`
 * Adds `lg2 checkout --force -- path1 path2` (currently only works as in actual `git` (i.e. without warning about conflicts) with `--force`)
 * Makes `lg2 status` output mention `lg2` client, rather than `git` client
---
### Refusal to create empty commits
Sample output on empty commit:
```
$ lg2 commit -m "commit: Refuse to create empty commits"
Error: No staged changes (nothing would be in the commit!). Refusing to commit.
Try running:
 *	lg2 status
	    to see which changes are staged (ready to commit) and which are unstaged.
 *	lg2 add path/to/changed/file
	    to stage a file.

Warning: An error occurred!
```

### `lg2 rm --cached`
 * `lg2 rm` (without `--cached`) would have to update the working tree
 * I would prefer to avoid `POSIX` file/directory APIs as much as possible for now :)
     * When `--cached` isn't given, instructions are given for removing the files from the working tree (with the `rm` shell command).

### `lg2 add --force`
 * Adds files even if in the `.gitignore`
 * It seems that [`git_index_update_all`](https://libgit2.org/libgit2/#HEAD/group/index/git_index_update_all) always ignores files in the `.gitignore`. `git_index_update_all` is used instead of `git_index_add_all` when `-u` is given.
     * As such, `--force` doesn't work when `-u` is given.

### CODE REVIEW

**Comment by personalizedrefrigerator** [comment]
This also fixes an issue caused by a dangling else:

https://github.com/holzschu/libgit2/blob/c23038d45db59fac308a23b2c71f3bb933f8890d/examples/commit.c#L84-L87

Initial commits in repositories were, as a result, not create-able because the `else` branch was not taken here and so `index` was never initialized.

File: examples/checkout.c [hunk 1]
```diff
@@ -260,49 +318,58 @@ int lg2_checkout(git_repository *repo, int argc, char **argv)
...
-	}
 
-	if (match_arg_separator(&args)) {
-		/**
-		 * Try to checkout the given path
-		 */
+		if (state == GIT_REPOSITORY_STATE_MERGE) {
+			fprintf(stderr, "It looks like a merge is in progress. Either "
+				"resolve the conflicts (see `lg2 status`), `lg2 add` each changed "
+				"file and commit the result, or run `lg2 reset HEAD` to stop the "
```
Comment (personalizedrefrigerator) [inline examples/checkout.c:332]: This might not be the correct way to cancel a merge... At present, 'lg2 merge --abort' is unimplemented.

**Comment by holzschu** [comment]
Thanks for this hard work. I'm going to make an a-Shell release with all this.

**Comment by holzschu** [comment]
Build 175 is now available, with this PR incorporated.

**Pull request merged**